guides/svg/template.md

```diff
@@ -0,0 +1,228 @@
+  
+    <svg id="icon-{{id}}" class="component-icon" width="32" height="32"></svg>
+
+    <!-- Metadata -->
+    <text class="component-name" y="1em">{{ name }}</text>
+    <text class="component-type" y="2.5em">{{ definition.type }}</text>
+</svg>
+```
+
+### 5. Add the container ***(only for container component)*** :
+To add a container, you need to add a `<svg></svg>` with a `component-container` class, and inside it, you need to add
+a `<rect></rect>` with a `container-background` class.
+
+**Important :** you need to set the `overflow` attribute to `visible` to be able to see the children when you drag them
+outside the container. You also need to fill the `container-background` with the `fill` attribute for drop detection.
+
+```jsx
+<svg class="template" width="250" height="200">
+    <rect class="component-hitbox" width="100%" height="100%" fill="white" />
+  
+    <svg id="icon-{{id}}" class="component-icon" width="32" height="32"></svg>
+
+    <text class="component-name" y="1em">{{ name }}</text>
+    <text class="component-type" y="2.5em">{{ definition.type }}</text>
+
+    <!-- Container -->
+    <svg class="component-container" overflow="visible">
+        <rect class="container-background" width="100%" height="100%" fill="whitesmoke"/>
+    </svg>
+</svg>
+```
+
+### 6. Add the error icon :
+This step is not mandatory, but if you want to display an error icon when the component has an error, you need to add
+a `<g></g>` with a `component-error` class.We used `nunjucks` to have a conditional rendering, if the component has an
+error, we display the error icon, if not, we don't display anything.
+
+**Important :** you need to set the `viewBox` attribute to be equal to your icon width and height.
+
+```jsx
+<svg class="template" width="250" height="200">
+    <rect class="component-hitbox" width="100%" height="100%" fill="white" />
+  
+    <svg id="icon-{{id}}"
+         class="component-icon"
+         width="32" height="32"></svg>
+
+    <text class="component-name" y="1em">{{ name }}</text>
+    <text class="component-type" y="2.5em">{{ definition.type }}</text>
+
+    <svg class="component-container" overflow="visible">
+        <rect class="container-background" width="100%" height="100%" fill="whitesmoke"/>
+    </svg>
+
+    <!-- Error icon-->
+    {% if hasError %}
+    <g class="component-error">
+        <!-- This is a placeholder, you can define the icon you want by replacing this svg. -->
+        <svg width="15" height="15" viewBox="0 0 100 100">
+            <circle fill="red" r="50" cx="50" cy="50"></circle>
+        </svg>
+    </g>
+    {% endif %}
+</svg>
+```
+
+### 7. Position the elements :
+Now that we have all the elements, we can position them.
+We can't use the `x` and `y` attributes, because the FireFox browser have weird behavior with them, so we use the
+`<g></g>` element and the `transform` attribute with the `translate(x,y)` function, to have a new position repository
```

Comment:
don't use repository. Suggestions :
position reference frame / position frame of reference / positional reference system or else.

Reply:
Done